test/TestProjects/SubClients-LowLevel/Generated/RootClient.cs

```diff
@@ -0,0 +1,135 @@
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="cachedParameter"/> or <paramref name="credential"/> is null. </exception>
+        public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, LlcSubClientsClientOptions options = null)
+        {
+            if (cachedParameter == null)
+            {
+                throw new ArgumentNullException(nameof(cachedParameter));
+            }
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            endpoint ??= new Uri("http://localhost:3000");
+            options ??= new LlcSubClientsClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _cachedParameter = cachedParameter;
+            _endpoint = endpoint;
+        }
+
+        /// <param name="context"> The request context. </param>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> GetCachedParameterAsync(RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("RootClient.GetCachedParameter");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateGetCachedParameterRequest();
+                return await _pipeline.ProcessMessageAsync(message, _clientDiagnostics, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="context"> The request context. </param>
+#pragma warning disable AZC0002
+        public virtual Response GetCachedParameter(RequestContext context = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("RootClient.GetCachedParameter");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateGetCachedParameterRequest();
+                return _pipeline.ProcessMessage(message, _clientDiagnostics, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private volatile Parameter _cachedParameter0;
+
+        /// <summary> Initializes a new instance of Parameter. </summary>
+        public virtual Parameter GetParameterClient()
```

Comment:
Can we add some unit tests to verify that the generated code is correct?

Reply:
Done